t/01-pdk/11-ctx.t

```diff
@@ -65,3 +65,69 @@ GET /t
 world
 --- no_error_log
 [error]
+
+
+
+=== TEST 3: ctx.plugin create namespace
+--- http_config eval: $t::Util::HttpConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            local PDK = require "kong.pdk"
+            local pdk = PDK.new()
+            local plugin = { name = "plugin" }
+
+            pdk.ctx["plugin"] = plugin
+
+            pdk.ctx.plugin.hello = "world"
+            pdk.ctx.plugin.cats = {
+              "marry",
+              "suzie"
+            }
+
+            ngx.say(pdk.ctx.plugin.hello)
+            ngx.say(pdk.ctx.plugin.cats[1])
+            ngx.say(ngx.ctx.plugin)
+            ngx.say(ngx.ctx.hello)
+        }
+    }
+--- request
+GET /t
+--- response_body
+world
+marry
+nil
+nil
+--- no_error_log
+[error]
```

Comment:
This doesn't seem like a relevant test to me. What I had in mind was a PDK instance and two light threads that would set a namespace, yield, and verify that resuming the first lighthread still ensures that the `pdk.ctx.plugin` namespace is preserved. Pretty much what happens when observing the issue in Kong itself.

Reply:
Issue is related to where `pdk.ctx.plugin` is stored, the namespace is preserved on both.

two light threads test works on `master` and `fix/pdk-race`. 
As fix introduce `ngx.ctx` store for `pdk.ctx.plugin`, the test should be against two concurrent requests (different `ngx.ctx`), to be sure `pdk.ctx.plugin` is not preserved.

what do you think ?